aggregator_core/src/datastore/test_util.rs

```diff
@@ -261,12 +261,33 @@ impl Drop for EphemeralDatastore {
         // Make a best-effort attempt to delete the database created for this datastore. This may
         // fail if the container hosting the database server is deleted faster, which would make
         // this moot.
-        let pool = self.pool.clone();
         let db_name = self.db_name.clone();
-        spawn(async move {
-            if let Ok(conn) = pool.get().await {
-                let _ = conn.execute("DROP DATABASE $1", &[&db_name]).await;
-            }
+        let connection_string = self.db.connection_string("postgres");
+
+        // Since each unit test has its own Tokio runtime, we need to run `DROP DATABASE` in a
+        // separate thread and Tokio runtime. If we just spawned a new task to do so in the same
+        // runtime, it would get cancelled when the runtime shuts down.
+        std::thread::spawn(move || {
+            let runtime = tokio::runtime::Builder::new_multi_thread()
+                .worker_threads(2)
```

Comment:
It doesn't really matter, but any particular reason that a current_thread runtime wouldn't work here?

Reply:
A current thread runtime would have worked when using one runtime per database, but it doesn't work anymore after I rewrote it to spawn fewer threads. The current thread runtime only makes progress if at least one thread is calling `block_on()`, but there's no longer a natural place to do so. The long-lived cleanup thread instead blocks on reading from a `std::sync::mpsc` channel, then spawns async tasks. In turn, I'm using this channel instead of `tokio::sync::mpsc` because `drop()` is a synchronous function called within a Tokio context, so `tokio::sync::mpsc::Sender::blocking_send()` is off-limits.